pkg/worker/k8srunner.go

```diff
@@ -123,8 +136,12 @@ func (f k8sStepRunnerFactory) NewStepRunner(
 			WithString("step", r.step.Name).
 			WithString("pod", r.target.name)
 	}
-	if err := r.dryRunStepError(ctx); err != nil {
-		return nil, fmt.Errorf("dry-run: %w", err)
+	if r.DryRun != DryRunNone {
+		// We skip running dry-run here, as it will be run later in the actual
+		// step run. Otherwise it would dry-run twice
```

Comment:
nit: end sentences with `.`
```suggestion
		// We skip running dry-run here, as it will be run later in the actual
		// step run. Otherwise it would dry-run twice.
```